### PR TITLE
DOCSP-27090 Add Atlas only stage note

### PR DESCRIPTION
**Pull request by ianf-mongodb**
## DESCRIPTION

Adding callout that Additional stages `$search` and `$serchMeta` are available when a user connects compass to Atlas. 

## STAGING

https://docs-mongodbcom-staging.corp.mongodb.com/compass/docsworker-xlarge/DOCSP-27090/aggregation-pipeline-builder/

## JIRA
https://jira.mongodb.org/browse/DOCSP-27090

## BUILD LOG

https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=63a475c8dc742b91f284ed12

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Is this free of spelling errors?
- [x] Is this free of grammatical errors?
- [x] Is this free of staging / rendering issues?
- [x] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/aggregation-pipeline-builder.txt [hunk 1]
```diff
@@ -31,6 +31,13 @@ aggregation pipeline. The :guilabel:`Preview of Documents in the
 Collection` section shows 20 documents randomly sampled from the chosen
 collection.
 
+.. note:: 
+
+   When you connect Compass to MongoDB instances hosted on
```
Comment (jeff-allen-mongo) [inline source/aggregation-pipeline-builder.txt:36]: [nit]

I'm admittedly forgetting the context around this decision, but I think we want to use "deployment" rather than "instance".

 ```suggestion
   When you connect Compass to a MongoDB deployment hosted on 
```

Comment (jeff-allen-mongo) [inline source/aggregation-pipeline-builder.txt:36]: [suggestion]

Use the replacement `|compass-short|` rather than "Compass".

File: source/aggregation-pipeline-builder.txt [hunk 2]
```diff
@@ -31,6 +31,13 @@ aggregation pipeline. The :guilabel:`Preview of Documents in the
 Collection` section shows 20 documents randomly sampled from the chosen
 collection.
 
+.. note:: 
+
+   When you connect Compass to MongoDB instances hosted on 
+   `Atlas <https://www.mongodb.com/cloud/atlas?tck=docs_compass>`_,
+   additional Atlas only stages ``$search`` and ``$searchMeta`` are
```
Comment (jeff-allen-mongo) [inline source/aggregation-pipeline-builder.txt:38]: I think "Atlas only" needs a hyphen.

```suggestion
   additional Atlas-only stages ``$search`` and ``$searchMeta`` are 
```

Comment (jeff-allen-mongo) [inline source/aggregation-pipeline-builder.txt:38]: [suggestion]

Can we provide links to the $search and $searchMeta stages:

- [$search](https://www.mongodb.com/docs/atlas/atlas-search/query-syntax/#-search)
- [$searchMeta](https://www.mongodb.com/docs/atlas/atlas-search/query-syntax/#-searchmeta)

We can use the `:atlas:` role to simplify the syntax for these links.

File: source/aggregation-pipeline-builder.txt [hunk 3]
```diff
@@ -31,6 +31,13 @@ aggregation pipeline. The :guilabel:`Preview of Documents in the
 Collection` section shows 20 documents randomly sampled from the chosen
 collection.
 
+.. note::
```
Comment (jeff-allen-mongo) [inline source/aggregation-pipeline-builder.txt:34]: [suggestion]

For findability, suggest giving this note a title like "Atlas Search Stages"

```suggestion
.. note:: Atlas Search Stages
```

Comment (jeff-allen-mongo) [inline source/aggregation-pipeline-builder.txt:34]: Consider placing this note in the "Create an Aggregation Pipeline" section, possibly at the end of step 1 where we instruct users to choose a stage.

**Review by jeff-allen-mongo — CHANGES_REQUESTED** [review]
@ianf-mongodb thanks for these edits. Nothing major here, just a few small comments re formatting / wording.

File: source/aggregation-pipeline-builder.txt (hunk 2 above)
Comment (ianf-mongodb) [reply source/aggregation-pipeline-builder.txt]: Sure thing, I added these as links and used the :atlas: role.

File: source/command-line-options.txt [hunk 4]
```diff
@@ -103,7 +103,7 @@ overrides the value in the :guilabel:`Settings` panel.
 .. option:: --protectConnectionStrings
 
    Hide credentials in connection strings. Passwords in connection 
-   strings are displayed as `*****`. 
+   strings are displayed as ``*****``.
```
Comment (ianf-mongodb) [inline source/command-line-options.txt:106]: Unrelated edit to fix a build warning.

**Review by ianf-mongodb — COMMENTED** [review]
@jeff-allen-mongo This is ready for another look!

File: source/aggregation-pipeline-builder.txt [hunk 5]
```diff
@@ -31,6 +31,15 @@ aggregation pipeline. The :guilabel:`Preview of Documents in the
...
 collection.
 
+.. note:: Atlas Search Stages
+
+   When you connect |compass-short| to a MongoDB deployment hosted on 
+   `Atlas <https://www.mongodb.com/cloud/atlas?tck=docs_compass>`_,
+   additional Atlas-only stages 
+   :atlas:`$search </reference/atlas-search/query-syntax/#-search>` and
+   :atlas:`$searchMeta </reference/atlas-search/query-syntax/#-searchmeta>`
+   are available in the Aggregation Pipeline Builder.
```
Comment (jeff-allen-mongo) [inline source/aggregation-pipeline-builder.txt:41]: [suggestion]

Maybe add a sentence explaining when you might use these stages:

"Use these stages to perform full-text search on Atlas collections."

or something

**Review by jeff-allen-mongo — APPROVED** [review]
LGTM % a final suggestion.

**Review by mabaasit — APPROVED** [review]
